web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -43,6 +43,10 @@ export function useDocumentTerminal(doc: types.DocumentTerminal) {
   const ctx = useAppContext();
   const { documentsService } = useWorkspaceContext();
   const [attempt, startTerminal] = useAsync(async () => {
+    if ('status' in doc) {
+      documentsService.update(doc.uri, { status: 'connecting' });
```

Comment:
In a situation where `DocumentGatewayCliClient` times out when waiting for the gateway to be created, it updates the status to `error`. When the gateway ends up being created, we want to update the state back to `connecting` when we attempt to connect with the client.

This is what this call accomplishes.